sympy/series/formal.py

```diff
@@ -812,6 +813,46 @@ def _compute_fps(f, x, x0, dir, hyper, order, rational, full):
 
     result = None
 
+    # flag for indicating symbolic terms in a function
+    fps_sym = False
+
+    symb, res = S.Zero, S.Zero
+    if isinstance(f, Mul):
+        nterm = S.One
+        for term in Mul.make_args(f):
+            if isinstance(term, Pow):
+                # for x**n symbolic terms
+                if term.exp.is_symbol:
+                    symb = term.exp
+                    fps_sym = True
+                # if power is an integer
+                elif sympify(term.exp).is_integer:
+                        nterm *= term
+
+                # for x**(n-2) or x**(n/2) symbolic terms
+                else:
+                    a, b, subt = S.Zero, S.Zero, S.Zero
+                    if isinstance(term.exp, Add):
+                        a, b = Add.make_args(term.exp)
+                        if a.is_symbol or b.is_symbol:
+                            if a.is_symbol:
+                                subt = a
+                            else:
+                                subt = b
+                            res = term.exp - subt
+                            if sympify(res).is_integer:
+                                nterm *= x**res
+                            symb = subt
+                            fps_sym = True
+
+                    if isinstance(term.exp, Mul):
+                        symb = term.exp
+                        fps_sym = True
+            else:
+                if not (isinstance(term, Pow) and term.exp.is_symbol):
```

Comment:
As far as I can figure out, control will reach `else` part only if, `isinstance(term, Pow)` is `False`. Now, `if not (isinstance(term, Pow) and term.exp.is_symbol)` can be interpreted as, `if not isinstance(term, Pow) or not term.exp.is_symbol` by DeMorgan's Laws and this will always be `True` as `not isinstance(term, Pow)` will always be `True`. So, overall, the check in L852 can be removed. Am I right?

Reply:
Hi @czgdp1807 thanks for analyzing this.

Hi @ArighnaIITG 

How about using something like this?

```
>>> from sympy import symbols, sin
>>> x, n = symbols('x n')
>>> e = x**n*sin(x**2)
>>> e.as_coeff_mul(x)
(1, (x**n, sin(x**2)))
```

I feel using something like this can greatly simplify it and not have you implement a custom term separation loop.

Reply:
On further thoughts,

You can continue this
```
>>> _, (t1, t2) = e.as_coeff_mul(x)
>>> fs = e.free_symbols
{x, n}
# for fps(e, x) free symbols will be fs - {x}, eg. here {n}
>>> t1.as_base_exp()
(x, n) # maybe presence of free symbols in exponential can be a good indication.
```